lib/AST/LocalizationFormat.cpp

```diff
@@ -154,19 +169,25 @@ readYAML(llvm::yaml::IO &io, T &Seq, bool, Context &Ctx) {
     if (io.preflightElement(i, SaveInfo)) {
       DiagnosticNode current;
```

Comment:
Could you please move this declaration to right before the `for` loop? The idea is to re-use stack space here for all of the elements.